tests/role_based_account_test.go

```diff
@@ -407,6 +408,173 @@ func TestAccountUpdateRoleBasedNil(t *testing.T) {
 	}
 }
 
+// TestAccountUpdateRoleBasedLegacy tests TxInternalDataAccountUpdate with the following scenario:
+// 1. Update key to RoleBasedKey having a LegacyKey for all roles
+// 2. Test RoleTransfer of the RoleBasedKey
+// 3-1. Test RoleUpdate of the RoleBasedKey
+// 3-2. Recover the updated account key to the previously used RoleBasedKey
+// 4. Test RoleFeePayer of the RoleBasedKey
+func TestAccountUpdateRoleBasedLegacy(t *testing.T) {
```

Comment:
Does this test check the both `true` and `false` cases from `AccountKeyLegacy.Validate`?

Reply:
I added two more cases checking `false` return.